liquidity-sources/src/sources/uniswap_v3/pool_fetching.rs

```diff
@@ -27,43 +28,26 @@ pub struct PoolInfo {
 }
 
 /// Pool state in a format prepared for solvers.
-#[derive(Clone, Debug, PartialEq, Serialize)]
+#[serde_as]
+#[derive(Clone, Debug, Default, PartialEq, Serialize, Deserialize)]
 pub struct PoolState {
-    #[serde(with = "serde_with::rust::display_fromstr")]
+    #[serde(with = "u256_decimal")]
     pub sqrt_price: U256,
-    #[serde(with = "serde_with::rust::display_fromstr")]
+    #[serde(with = "u256_decimal")]
     pub liquidity: U256,
     #[serde(with = "serde_with::rust::display_fromstr")]
     pub tick: BigInt,
-    pub liquidity_net: TickInfo,
+    // (tick_idx, liquidity_net)
+    #[serde_as(as = "BTreeMap<DisplayFromStr, DisplayFromStr>")]
+    pub liquidity_net: Vec<(BigInt, BigInt)>,
```

Comment:
I didn't know serde_as would convert from Vec to BTreeMap, interesting. why is it needed though?

Reply:
Essentially, `BTreeMap` requires less json space since it's serialized as a map. You can check the unit test `encode_decode_pool_info` what is the expected json format for `liquidity_net` field.

Reply:
Found the example here: https://docs.rs/serde_with/2.0.0/serde_with/#advanced-serde_as-usage

Reply:
Can the field  contain duplicate keys? If it cannot then BTreeMap makes sense.